src/Symfony/Component/Config/Definition/Builder/ExprBuilder.php

```diff
@@ -77,7 +77,7 @@ public function ifTrue(\Closure $closure = null): static
     public function ifString(): static
     {
         $this->ifPart = \is_string(...);
-        $this->allowedTypes = self::TYPE_STRING;
+        $this->allowedTypes = self::TYPE_STRING; // XXX why is this used by ConfigBuilderGenerator instead of the prototype?
```

Comment:
By default it's type `TYPE_ANY` what results in `mixed`. For me it seems that this is a workaround to interpret this as a string after the check.